tests/fixtures/upgrade/Cargo.toml.lockfile_target

```diff
@@ -0,0 +1,43 @@
+[package]
+name = "None"
+version = "0.1.0"
+
+[lib]
+path = "dummy.rs"
+
+[dependencies]
+docopt = "0.8.3"
+pad = "0.1.5"
+serde_json = "1.0.40"
+syn = { version = "0.11.11", default-features = false, features = ["parsing"] }
+tar = { version = "0.4.26", default-features = false }
+ftp = "2.2.1"
+
+[dependencies.semver]
+features = ["serde"]
+version = "0.7.0"
+
+[dev-dependencies]
+assert_cli = "0.2.2"
+tempdir = "0.3.7"
+
+[build-dependencies]
+serde = { version = "1.0", git= "https://github.com/serde-rs/serde.git" }
```

Comment:
I feel a bit uneasy with all the dependencies here, why is it not enough to have the same dependencies as in  `tests/fixtures/workspace/one/Cargo.toml`?
> The local lock file must be present and up to date if
this flag is passed.

Does this mean CI will fail each time a new version is released (or git commit is pushed)?

Reply:
I used cargo's terminology here: a lockfile is up-to-date if it satisfies the Cargo.toml, not if all the dependencies are at the most recent semver compatible version. See docstring for `--locked` at https://doc.rust-lang.org/cargo/commands/cargo-metadata.html.

It's a little confusing, but I thought it better to be consistent with cargo. WDYT?

The network isn't accessed when the `--to-lockfile` flag is passed, so CI won't fail. I've simplified the test as you suggested, and deliberately added a lockfile with an older version of `rand` to demonstrate.